### DISCUSSION TITLE
Code Annotations with C#

### DISCUSSION THREAD

**Discussion by DerTomDerTwitch** [squidfunk/mkdocs-material#2792] [Q&A] [answered]
I am trying to get code annotations to work with C#.  The example from the docs (written in JavaScript) works fine, but I cannot get it to work with C# code. 

```` { csharp .annotate }
using System; (1)

namespace HelloWorld
{
    public class HelloWorld
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
            Console.ReadKey();
        }
    }
}
1. This is a test`
````
(And of course surrounded by the usual 3 markdown ticks.

I tried to change the C# identifier according to the pygments lexers documentation (I tried "csharp", "c#" and also some random ones like ".cs") but cannot get it to work.

**Comment by squidfunk** [comment, marked as answer]
Annotations must be encapsulated inside comments, so I guess for C# that is either `// (1)` or `/* (1) */` 😊 Also, not that you must set the `.annotate` class explicitly, so a slightly different prelude syntax is necessary:

````
``` { .csharp .annotate }
your code
```
````

Also see [the example in the documentation](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/?h=annot#adding-annotations).

**Reply by DerTomDerTwitch** [reply]
I'm pretty sure I tested it exactly as you proposed yesterday evening, but since it simply works now. It must have been a lack of coffee. Was a long day ;-)

**Reply by DerTomDerTwitch** [reply]
Ah, nevermind. I tried `.cs`, `c#` and `csharp` but not `.csharp`

**Reply by squidfunk** [reply]
No worries, glad it works now!
